cmd/addvalidators_internal_test.go

```diff
@@ -118,33 +118,79 @@ func TestRunAddValidators(t *testing.T) {
 		n        = 3
 		valCount = 1
 	)
-	lock, p2pKeys, _ := cluster.NewForT(t, valCount, n, n, 0)
-
-	tmp := t.TempDir()
-	manifestFile := path.Join(tmp, "cluster-manifest.pb")
-
-	conf := addValidatorsConfig{
-		NumVals:           1,
-		WithdrawalAddrs:   []string{feeRecipientAddr},
-		FeeRecipientAddrs: []string{feeRecipientAddr},
-		TestConfig: TestConfig{
-			Lock:    &lock,
-			P2PKeys: p2pKeys,
-		},
-		ManifestFile: manifestFile,
-	}
 
-	err := runAddValidatorsSolo(context.Background(), conf)
-	require.NoError(t, err)
+	t.Run("add validators once", func(t *testing.T) {
+		lock, p2pKeys, _ := cluster.NewForT(t, valCount, n, n, 0)
+
+		tmp := t.TempDir()
+		manifestFile := path.Join(tmp, "cluster-manifest.pb")
+
+		conf := addValidatorsConfig{
+			NumVals:           1,
+			WithdrawalAddrs:   []string{feeRecipientAddr},
+			FeeRecipientAddrs: []string{feeRecipientAddr},
+			TestConfig: TestConfig{
+				Lock:    &lock,
+				P2PKeys: p2pKeys,
+			},
+			ManifestFile: manifestFile,
+		}
+
+		err := runAddValidatorsSolo(context.Background(), conf)
+		require.NoError(t, err)
+
+		// Verify the new cluster manifest
+		b, err := os.ReadFile(manifestFile)
+		require.NoError(t, err)
+
+		var msg manifestpb.Cluster
+		require.NoError(t, proto.Unmarshal(b, &msg))
+		require.Equal(t, len(msg.Validators), 2) // valCount+1
+		require.Equal(t, msg.Validators[1].FeeRecipientAddress, feeRecipientAddr)
+	})
+
+	t.Run("add validators twice", func(t *testing.T) {
```

Comment:
Note that this test can be easily generalized to test that `n` consecutive runs actually add `n` new validators to the cluster manifest. I have just considered `n == 2` for simplicity